BREAKING CHANGE: OIDC depreciated dependency switch to current (#2733)
unoplatform/uno.extensions: doc/Learn/UpdatingExtensions.md

# Upgrading Extensions Version

## Upgrading to Extensions 7.0

### OidcClient Authentication

When upgrading to Uno.Extensions 7.0 or later, the NuGet Package Dependency, before known as `IdentityModel.OidcClient`, which is used in the [Oidc Authentication Extension](xref:Uno.Extensions.Authentication.HowToOidcAuthentication), has been [rebranded](https://github.com/DuendeSoftware/foss/blob/main/README.md#relationship-to-identitymodel).

When upgrading to later versions, you must make sure to update the Namespaces in your App, to match the new ones. e.g. in your `GlobalUsings.cs` file at your Project root directory:

```diff
- global using IdentityModel.OidcClient
+ global using Duende.IdentityModel.OidcClient
```

> In case you only using the `Uno.Extensions.Authentication.Oidc` pulled in through the `<UnoFeatures>AuthenticationOidc</UnoFeatures>` in your `.csproj` and don't have any extra namespace usings added, there would be no further change required.

Reviewer: This paragraph I think we can omit, the namespace change above is enough

The Uno SDK dependencies for Uno Extensions have been updated to Uno SDK version 6.0. Ensure to [update your uno.sdk](xref:Uno.Development.UpgradeUnoNuget) to the latest version.

### OidcClient Authentication

Up from the v6.0.12 of Uno.Extensions, the NuGet Package Dependency known as `IdentityModel.OidcClient`, used in the [Oidc Authentication Extension](xref:Uno.Extensions.Authentication.HowToOidcAuthentication), has been [rebranded](https://github.com/DuendeSoftware/foss/blob/main/README.md#relationship-to-identitymodel).

> When upgrading to later versions, you must make sure to update the Namespaces in your App, to match the new ones. e.g. in your `globalUsings.cs` file at your Project root directory:

Reviewer: ```suggestion
When upgrading to later versions, you must make sure to update the Namespaces in your App, to match the new ones. e.g. in your `GlobalUsings.cs` file at your Project root directory:
```

```diff
- global using IdentityModel.OidcClient
+ global using Duende.IdentityModel.OidcClient
```

> In case you only using the `Uno.Extensions.Authentication.Oidc` pulled in through the `<UnoFeatures>AuthenticationOidc</UnoFeatures>` in your `.csproj` and don't have any future Namespace usings added, there is nothing left to do!

Reviewer: ```suggestion
In case you only using the `Uno.Extensions.Authentication.Oidc` pulled in through the `<UnoFeatures>AuthenticationOidc</UnoFeatures>` in your `.csproj` and don't have any extra namespace usings added, there would be no further change required.
```

> [!TIP]

> > To update eventual other Packages of this Provider like the `IdentityServer`, you may use alongside of the OidcClient, you can check the NuGet Package Manager in Visual Studio 2022, Rider or using the dotnet CLI if you may use VS Code.

Reviewer: As this will be part of the next release, which will be a major version bump, we should move this out of the Upgrading to Extensions 6.0 section and into a new Upgrading to Extensions 7.0 section above it

> > [!TIP]
> > To update eventual other Packages of this Provider like the `IdentityServer`, you may use alongside of the OidcClient, you can check the NuGet Package Manager in Visual Studio 2022, Rider or using the dotnet CLI if you may use VS Code.

Reviewer: ```suggestion
```
